src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -108,69 +106,30 @@ public static Tag parseTag(String tag) throws ParseException {
-     */
-    public static String parseAnswer(String answer) throws ParseException {
-        requireNonNull(answer);
-        String trimmedAnswer = answer.trim();
-        // TODO: Refactor this when the Answer/Options class is created:
-        //  Parse exception string should be in Answer/Options class, as well as the validation logic
-        //  (predicate in if statement)
-        if (trimmedAnswer.equals("")) {
-            throw new ParseException(MESSAGE_INVALID_ANSWER);
-        }
-        return trimmedAnswer;
-    }
-
-    /**
-     * Parses {@code String options} into a {@code String}.
-     * TODO: Change this to return Option Object instead
+     * Parses {@code String choice} into a {@code Choice choice}.
      */
-    public static String parseOption(String option) throws ParseException {
-        requireNonNull(option);
-        String trimmedOption = option.trim();
-        // TODO: Refactor this when the Option class is created:
-        //  Parse exception string should be in Option class, as well as the validation logic
-        //  (predicate in if statement)
-        if (trimmedOption.equals("")) {
-            throw new ParseException(MESSAGE_INVALID_OPTION);
+    public static Choice parseChoice(String choice, boolean isCorrect) throws ParseException {
+        requireNonNull(choice);
+        String trimmedChoice = choice.trim();
+        if (!Choice.isValidChoiceTitle(trimmedChoice)) {
+            throw new ParseException(Choice.MESSAGE_CONSTRAINTS);
         }
-        return trimmedOption;
+        return new Choice(trimmedChoice, isCorrect);
     }
 
     /**
-     * Parses {@code List<String> options} into a {@code String[]}.
-     * TODO: Change this to return Options Object instead
+     * Parses {@code List<String> choices, String answer} into a {@code Set<Choice> choice}.
      */
-    public static String[] parseOptions(List<String> options) throws ParseException {
-        requireNonNull(options);
-        // TODO: Refactor this when the Options class is created:
-        //  Parse exception string should be in Options class, as well as the validation logic
-        //  (predicate in if statement + cannot have duplicate options)
-        if (options.size() < 3) {
-            throw new ParseException(MESSAGE_INSUFFICIENT_OPTIONS);
+    public static Set<Choice> parseChoices(List<String> choices, String answer) throws ParseException {
+        requireAllNonNull(choices, answer);
+        if (choices.size() < NUMBER_OF_INCORRECT_CHOICES) {
+            throw new ParseException(MultipleChoiceQuestion.MESSAGE_INSUFFICIENT_CHOICES);
         }
-        String[] optionsArr = options.stream().limit(3).toArray(String[]::new);
-        for (int i = 0; i < optionsArr.length; i++) {
-            optionsArr[i] = parseOption(optionsArr[i]);
+        Set<Choice> choiceSet = new HashSet<>();
+        for (int i = 0; i < NUMBER_OF_INCORRECT_CHOICES; i++) {
+            choiceSet.add(parseChoice(choices.get(i), false));
```

Comment:
Shall we add another check here to ensure that number of  choices is still 3? If not, user input mcq ... opt/1 opt/1 opt/1 ans/2 will pass the check but in the end there will only be one incorrect option. The message for this can be something like "Options should not be duplicates".

Reply:
This is a nitpick and probably wouldn't happen with actual users but we're also trying to defend against people intentionally breaking our app haha

Reply:
Okay! I have added the check